adapter/internal/operator/operator.go

```diff
@@ -67,11 +68,12 @@ func init() {
 }
 
 // InitOperator starts the Kubernetes gateway operator
-func InitOperator() {
+func InitOperator(prometheusPort int32) {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	port := strconv.FormatInt(int64(prometheusPort), 10)
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":"+port, "The address the metric endpoint binds to.")
```

Comment:
This line should be governed from the metrics configuration you have in the config.toml.

Reply:
Fixed in [18fc290](https://github.com/wso2/apk/pull/2135/commits/18fc290c1e2e64c42ef33ff55697a3d9139bb8bd)